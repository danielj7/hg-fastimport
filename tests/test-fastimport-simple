#!/bin/sh

# A simple dump: two commits, no marks, no blobs

. $TESTDIR/fastimport-common

cat > simple.dump <<__EOF__
commit refs/heads/master
committer Example <example@example.org> 1200000000 +0000
data 17
initial revision

M 644 inline file1
data 14
file 1 begins

M 644 inline file2
data 19
this here be file2

commit refs/heads/master
committer Example <example@example.org> 1200000001 +0000
data 11
add a line

M 644 inline file1
data 23
file 1 begins
new line

__EOF__

set -e
rm -rf simple
echo "% importing"
hg init simple
cd simple
hg fastimport ../simple.dump | sed "s|$HGTMP|HGTMP|g"

# It's OK for the log to include changeset IDs, since the fastimport
# dump includes everything used to compute them, so they should be the
# same from run to run.
echo "% hg log"
hg log

echo "% content of file1, rev 0"
hg cat -r0 file1

echo "% content of file2, rev 0"
hg cat -r0 file2

echo "% content of file1, rev 1"
hg cat -r1 file1

echo "% content of file2, rev 1"
hg cat -r1 file2
