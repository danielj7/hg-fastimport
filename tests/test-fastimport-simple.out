% importing
scanning source...
sorting...
converting...
1 initial revision
0 add a line
Removing blob dir 'HGTMP/test-fastimport-simple/simple/.hg/blobs' ...
% hg log
changeset:   1:ec2bd7ac5b27
tag:         tip
user:        Example <example@example.org>
date:        Thu Jan 10 21:20:01 2008 +0000
summary:     add a line

changeset:   0:b7a8255f9622
user:        Example <example@example.org>
date:        Thu Jan 10 21:20:00 2008 +0000
summary:     initial revision

% content of file1, rev 0
file 1 begins
% content of file2, rev 0
this here be file2
% content of file1, rev 1
file 1 begins
new line
% content of file2, rev 1
this here be file2
